### ISSUE TITLE
How to tell if the current ScreenView is currently active?

### ISSUE THREAD

**Issue by jonathanolson** [phetsims/joist#536]
It seems the ScreenView doesn't have a reference to the Screen (which has an activeProperty), and Sim's currentScreenProperty is joist-only.

I'm going to temporarily "break" the joist-only bit to temporarily check `phet.joist.sim.currentScreenProperty.value === screenView`, but I'm flagging this to determine if we should make that permanent (remove 'joist-only'), or come up with a better way.

**Comment by jessegreenberg** [comment]
I think we should have a better way to do this that doesn't go through the phet namespace.

**Comment by pixelzoom** [comment]
Yeah, what's the use case?  Why does the ScreenView need to know if it's active?  

And if it does need to know, create `{BooleanProperty} isActiveProperty` in Screen and pass it to its ScreenView.

**Comment by jonathanolson** [comment]
> I don't understand why the BuildingLabScreenView.inputListener needs to know which screen is active.

There is a listener added on the Display to catch input events in the "background" (e.g. like ComboBox) that, if no relevant object was interacted with, will "unselect" everything. We want to start with one thing selected on the screen.

The problem is when the user starts on the home screen and presses on the screen button, that activates our listener and deselects the initial object.

**Comment by samreid** [comment]
Somewhat devil's advocating here, but want to understand better: Shouldn't a Display event in the "background" unselect everything in every screen?  Or are non-active screens expected to retain their selections?  Say I open a combo box on screen 1, then go to screen 2 and tap in the display.  When I return to screen 1, should I expect the combo box to still be open?

**Comment by zepumph** [comment]
Just a thought that breaks OO programming, but I thought I'd get it out here (we are all thinking it)

In ScreenView:
```
// the parent type
this.screen = screen
```

In usage
`if( this.screen.activeProperty.get() )`

**Comment by pixelzoom** [comment]
This seems to work just fine.  What doesn't this do that you need?

```js
// MyScreen.js
class MyScreen extends Screen {
    constructor() {
      super( 
        () => new MyModel(), 
        model => new MyScreenView( this.activeProperty, model ),
         {... } );
    }
  }

// MyScreenView.js
class MyScreenView extends ScreenView {
  constructor( isActiveProperty ) {
      ...
      isActiveProperty.link( isActive ) {
        // do something when the screen's activation changes
      }
  }
}
```

**Comment by pixelzoom** [comment]
First,  I thought we were concerned with how to tell if a ScreenView should be active here.  And passing `isActiveProperty` to its constructor means that the ScreenView doesn't care where that Property comes from or what's controlling it.  There is zero coupling between Sim, Screen and ScreenView, all of which are interested in which screen is active.

In terms of generality... Yes, if you wanted the ScreenView to observe something else (e.g. whether it should be visible or not), then I'd be advocating passing in another Property (e.g. visibleProperty).

**Comment by pixelzoom** [comment]
... and passing in `activeProperty` to ScreenView sure seems a lot better than resorting to a global variable, `check phet.joist.sim.currentScreenProperty.value === screenView`, as you're doing now.

**Comment by jonathanolson** [comment]
Passing in just the Property seems fine.

**Comment by pixelzoom** [comment]
@jonathanolson is there something else to discuss here for dev meeting, or can this be unlabeled?
